services:
  django_photo_app:
    container_name      : django_photo_app
    build: 
      context: .
      dockerfile: Dockerfile.django
    ports:
      - "8000:8000"
    volumes:
      - ./app:/app
    env_file:
      - .env
    depends_on:
      db:
        condition: service_healthy

  nginx-proxy:
    container_name      : nginx-proxy
    build:
      context: .
      dockerfile: Dockerfile.nginx
    image: nginx:alpine
    ports:
      - "80:80"
    volumes:
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
      - static_volume:/app/static
    depends_on:
      - django_photo_app

  db:
    image: postgres:17.5
    container_name: postgresql_db
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: ProGATE4623!
      POSTGRES_DB: local_postgredb
    ports:
      - "5432:5432"
    volumes:
      - pgdata:/var/lib/postgresql/data 
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres -d local_postgredb"]
      interval: 5s
      timeout: 5s
      retries: 5

volumes:
  pgdata:
  static_volume: